classes/ProductDownload.php

```diff
@@ -252,11 +252,12 @@ public static function getFilenameFromFilename($filename)
      */
     public function getTextLink($admin = true, $hash = false)
     {
+        if ($admin) {
+            return 'get-file-admin.php?file=' . $this->filename;
+        }
         $key = $this->filename . '-' . ($hash ? $hash : 'orderdetail');
-        $link = ($admin) ? 'get-file-admin.php?' : _PS_BASE_URL_ . __PS_BASE_URI__ . 'index.php?controller=get-file&';
-        $link .= ($admin) ? 'file=' . $this->filename : 'key=' . $key;
 
-        return $link;
+        return  _PS_BASE_URL_ . __PS_BASE_URI__ . 'index.php?controller=get-file&key=' . $key;
```

Comment:
it would be better to `Link` to generate the url
With `Link::getPageLink` maybe?